JitsiConference.js

```diff
@@ -116,6 +117,19 @@ JitsiConference.prototype._init = function (options) {
     // Always add listeners because on reload we are executing leave and the
     // listeners are removed from statistics module.
     this.eventManager.setupStatisticsListeners();
+
+    if (this.options.config.enableTalkWhileMuted) {
+        this.talkMutedDetection
+            = new TalkMutedDetection(() => {
+                this.eventEmitter.emit(JitsiConferenceEvents.TALK_WHILE_MUTED);
+            });
+        this.statistics.addAudioLevelListener(
+            this.talkMutedDetection.audioLevelListener
+                .bind(this.talkMutedDetection));
+        this.eventEmitter.on(
+            JitsiConferenceEvents.TRACK_MUTE_CHANGED,
+            this.talkMutedDetection.muteChanged.bind(this.talkMutedDetection));
```

Comment:
I think that `TalkMutedDetection` should receive this `JItsiConference` in its constructor, add the listeners it's interested in and, thus, move closer to a black box that `JitsiConference` merely activates by `new TalkMutedDetection`.